app/assets/javascripts/tinymce/plugins/custom_image_toolbar/plugin.js

```diff
@@ -0,0 +1,34 @@
+/* global tinymce MarvinJsEditor */
+tinymce.PluginManager.add('custom_image_toolbar', function(editor) {
+  editor.addButton('image_download', {
+    icon: 'download',
+    onclick: function() {
+      var editorIframe = $('#' + editor.id).prev().find('.mce-edit-area iframe');
+      var image = editorIframe.contents().find('img[data-mce-selected="1"]');
+      window.open('/tiny_mce_assets/' + image.data('mceToken') + '/download', '_blank');
```

Comment:
Can we put whole download URL to data attribute?

Reply:
I think it will increase logic because we need save multiple URL for each action, and after save, clean this data for each image.

Reply:
OK